src/systems/hydrodynamics/Hydrodynamics.cc

```diff
@@ -78,14 +79,136 @@ class gz::sim::systems::HydrodynamicsPrivateData
   /// \brief Previous state.
   public: Eigen::VectorXd prevState;
 
+  /// \brief Previous derivative of the state
   public: Eigen::VectorXd prevStateDot;
 
-  /// Link entity
+  /// \brief Use current table if true
+  public: bool useCurrentTable {false};
+
+  /// \brief Current table xComponent
+  public: std::string axisComponents[3];
+
+  public: std::shared_ptr<gz::sim::components::EnvironmentalData>
+    gridField;
+
+  public: std::optional<gz::math::InMemorySession<double, double>> session[3];
+
+  /// \brief Link entity
   public: Entity linkEntity;
 
   /// \brief Ocean current callback
   public: void UpdateCurrent(const msgs::Vector3d &_msg);
 
+  /////////////////////////////////////////////////
+  /// \brief Set the current table
+  /// \param[in] _ecm - The Entity Component Manager
+  /// \param[in] _currTime - The current time
+  public: void SetCurrentTable(
```

Comment:
@arjo129 nit^N: current could be understood as present. Consider qualifying it: ocean current, water current, you name it.